src/ipz_parser.cpp

```diff
@@ -625,4 +625,216 @@ types::DbusVariantType IpzVpdParser::readKeywordFromHardware(
     return types::DbusVariantType{
         getKeywordValueFromRecord(l_record, l_keyword, l_recordOffset)};
 }
+
+void IpzVpdParser::updateRecordECC(const auto& i_recordDataOffset,
+                                   const auto& i_recordDataLength,
+                                   const auto& i_recordECCOffset,
+                                   size_t i_recordECCLength,
+                                   types::BinaryVector& io_vpdVector)
+{
+    auto l_recordDataBegin = std::next(io_vpdVector.begin(),
+                                       i_recordDataOffset);
+
+    auto l_recordECCBegin = std::next(io_vpdVector.begin(), i_recordECCOffset);
+
+    auto l_eccStatus = vpdecc_create_ecc(
+        const_cast<uint8_t*>(&l_recordDataBegin[0]), i_recordDataLength,
+        const_cast<uint8_t*>(&l_recordECCBegin[0]), &i_recordECCLength);
+
+    if (l_eccStatus != VPD_ECC_OK)
+    {
+        throw(EccException("ECC update failed."));
+    }
+
+    auto l_recordECCEnd = std::next(l_recordECCBegin, i_recordECCLength);
```

Comment:
isn't this    l_recordECCEnd = std::next(l_recordECCBegin, i_recordECCLength);   same as 
l_recordECCEnd  = i_recordECCOffset + i_recordECCLength

wherever end offset  was required you could have just used start offset + length as they are constants coming in

similar to what is used in the below seekp

Reply:
I have not used m_vpdStartOffset
I have meant of using the start offset which came as parameter of the function , start offset does not mean file start offset , it could be record start offset or etc start offset depending on where it is used

Reply:
>>isn't this l_recordECCEnd = std::next(l_recordECCBegin, i_recordECCLength); same as
l_recordECCEnd = i_recordECCOffset + i_recordECCLength

yes we can do that also..  just made use of C++ STL here.